### PR TITLE
HW#1-OOP-Khokhlov Yevhenii

### PR DESCRIPTION
**Pull request by yevhenii-khokhlov**

### CODE REVIEW

File: .idea/.gitignore [hunk 1]
```diff
@@ -0,0 +1,3 @@
+# Default ignored files
+/shelf/
+/workspace.xml
```
Comment (dimigor) [inline .idea/.gitignore:3]: please, exclude all .idea files
you can create compete .gitignore file here: https://www.toptal.com/developers/gitignore

File: homework/Animals.py [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+        return self.color
+
+    @property
+    def get_size(self) -> str:
+        return self.size
+
+    def set_size(self, size: str):
+        self.size = size
+
+    def set_color(self, color: str):
```
Comment (dimigor) [inline homework/Animals.py:17]: you can also use `property.setter`

File: homework/Animals.py [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+        self.color = color
+
+
+class Dog(Animal):
+    def bark(self):
+        print('woof-woof')
+        return self
+
+
+class Wolf(Dog, color='grey'):
```
Comment (dimigor) [inline homework/Animals.py:27]: `color='grey'` what does it mean?

File: homework/composition.py [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+    def __init__(self, name: str, size: int):
+        self.name = name
+        self.size = size
+
+
+class Laptop:
+    def __init__(self):
+        app1 = App('Chrome', 500)
+        app2 = App('Photoshop', 4000)
+        app3 = App('TextRedactor', 42)
```
Comment (dimigor) [inline homework/composition.py:11]: create dict with values {'Chrome': 500, 'Photoshop': 4000} and in loop create instances

File: homework/metaclass.py [hunk 5]
```diff
@@ -0,0 +1,15 @@
+ParentMetaClass = type('ParentMetaClass', (), {'parent_class': True})
+
+MetaClass = type('MetaClass', (ParentMetaClass,), {'metaclass': True})
+
+
+class MyClass(MetaClass):
+    def my_class(self):
+        print('Hi from MyClass!')
+        return self
```
Comment (dimigor) [inline homework/metaclass.py:9]: you do not need to return self

**Review by dimigor — CHANGES_REQUESTED** [review]

File: homework/Animals.py (hunk 2 above)
Comment (yevhenii-khokhlov) [reply homework/Animals.py:17]: Done

File: homework/metaclass.py (hunk 5 above)
Comment (yevhenii-khokhlov) [reply homework/metaclass.py:9]: Done

File: homework/Animals.py (hunk 3 above)
Comment (yevhenii-khokhlov) [reply homework/Animals.py:27]: this is my mistake, 
I deleted

File: homework/composition.py (hunk 4 above)
Comment (yevhenii-khokhlov) [reply homework/composition.py:11]: Done

**Pull request closed without merging**